File: Ejercicios/G.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n;
    cin >> n;
    string bal;
    cin >> bal;

    vector<int> l = {0};
    int x = 0;
    for (char c : bal) {
        if (c == 'X') {
            x++;
            l.push_back(x);
        } else {
            l.push_back(x);
        }
    }

    int subsecuencias = 0;
    for (int i = 3; i <= sqrt(n); i++) {
        int xesperadas = 4*i - 4;
        int oesperadas = pow(i-2, 2);
        for (int j = 0; j <= n - i*i; j++) {
            int xs = l[j+i*i] - l[j];
            int os = i*i - xs;
            if (xs == xesperadas && os == oesperadas) {
                subsecuencias++;
            }
        }
    }

    cout << subsecuencias << "\n";
}

File: Ejercicios/G.py
from math import sqrt
n = int(input())
bal = input()
l = [0]
x = 0

for i in bal:
    if i == "X":
        x += 1
        l.append(x)
    else:
        l.append(x)
        
subsecuencias = 0

for i in range(3, int(sqrt(n)) + 1):
    xesperadas = ((2 * i) + 2*(i-2))
    oesperadas = ((i - 2) ** 2)

    for j in range((n - i**2) + 1):
        xs = (l[j + (i**2)] - l[j])
        os = i**2 - xs
        if xs == xesperadas and os == oesperadas:
            subsecuencias += 1

print(subsecuencias)